File: Unit_tests_python/EX-04-Calculate_stats/Claude.py
import statistics

def calculate_stats(list1, list2, list3):
    mean1 = statistics.mean(list1)
    mean2 = statistics.mean(list2) 
    mean3 = statistics.mean(list3)
    
    stddev1 = statistics.pstdev(list1)
    stddev2 = statistics.pstdev(list2)
    stddev3 = statistics.pstdev(list3)
    
    means = [mean1, mean2, mean3]
    stddevs = [stddev1, stddev2, stddev3]
    
    stats = {'mean': means, 'stddev': stddevs}
    return stats

File: Unit_tests_lua/EX-04-Calculate_stats/Claude.lua
local math = require('math')

function calculate_stats(list1, list2, list3)
  local means = {}
  local stddevs = {}
  
  for i, list in ipairs({list1, list2, list3}) do
    local sum = 0
    for _, x in ipairs(list) do
      sum = sum + x
    end
    local mn = sum / #list
    table.insert(means, mn)
    
    local var = 0
    for _, x in ipairs(list) do
      var = var + (x - mn)^2 
    end
    local std = math.sqrt(var / #list)
    table.insert(stddevs, std)
  end
  
  return {
    mean = means,
    stddev = stddevs
  }
end